### PR TITLE
Allow rollup@4.x.x as peer dependency

### PR DESCRIPTION
**Pull request by smholsen**
Spinning up this PR from https://github.com/vanilla-extract-css/vanilla-extract/discussions/1289.
@mrm007 I don't know exactly what to test/verify for this change though. 
Heres the breaking changes from 3.x.x to 4.x.x: https://github.com/rollup/rollup/releases/tag/v4.0.0

I've done a quick scan, and it seems theres a few references to `preserveModules` in a test, which I am unsure if might cause errors down the line if the peer dependency is on `4.x.x` due to the breaking change note:
```
inlineDynamicImports, manualChunks and preserveModules have been removed on input option level: Please use the corresponding output options of the same names (https://github.com/rollup/rollup/pull/5143)
```

Other than that it looks to me this should be safe. 
Anything else we need to verify for this?

### CODE REVIEW

File: .changeset/clean-wolves-attend.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@vanilla-extract/rollup-plugin': minor
```
Comment (mrm007) [inline .changeset/clean-wolves-attend.md:2]: Looks like we treat dependency bumps as `patch`, sorry about the confusion.

**Review by mrm007 — APPROVED** [review]
Thanks @smholsen!

**Comment by mrm007** [comment]
I thought it would be a good idea to bump `devDependencies` for the Rollup plugin. Unfortunately, [this line](https://github.com/vanilla-extract-css/vanilla-extract/blob/a0fd6234bfe8294639d039bd25ec7135cc3445aa/packages/rollup-plugin/src/index.ts#L104) causes a TypeScript error and the sourcemaps test failed in unexpected ways, so I'll do that in a separate PR.

**Comment by mrm007** [comment]
@graup do you see any issues with this PR?

**Comment by smholsen** [comment]
Thanks for the quick follow up here guys! 🙌

**Pull request merged**